app/src/features/FhirResourceTree/resourceTreeUtils.ts

```diff
@@ -199,51 +262,174 @@ export const getParent = (
   elementDefinitions: IElementDefinition[],
-  rootNode: ElementNode,
-  previousElementNode: ElementNode
+  rootNodeDefinition: DefinitionNode,
+  previousElementNodeDefinition: DefinitionNode
 ): void => {
   const [currentElementDefinition, ...rest] = elementDefinitions;
   if (!currentElementDefinition) return;
 
   if (isOmittedElement(currentElementDefinition)) {
-    buildTree(rest, rootNode, previousElementNode);
+    buildTreeDefinition(
+      rest,
+      rootNodeDefinition,
+      previousElementNodeDefinition
+    );
     return;
   }
 
-  const currentElementNode = createElementNode(currentElementDefinition, {
-    parentPath: rootNode.path,
-  });
+  const currentDefinitionNode: DefinitionNode = createDefinitionNode(
+    currentElementDefinition
+  );
 
-  if (currentElementNode.kind === "choice") {
-    currentElementNode.children = getChildrenChoices(
-      currentElementNode.definition,
-      rootNode.path
-    );
-  }
+  if (
+    isSliceOf(
+      currentElementDefinition,
+      previousElementNodeDefinition.definition
+    )
+  ) {
+    previousElementNodeDefinition.sliceDefinitions.push(currentDefinitionNode);
+    buildTreeDefinition(rest, rootNodeDefinition, currentDefinitionNode);
+  } else if (
+    isElementDefinitionChildOf(
+      currentElementDefinition,
+      previousElementNodeDefinition.definition
+    )
+  ) {
+    if (getKind(currentElementDefinition) === "choice") {
+      currentDefinitionNode.childrenDefinitions = getChildrenChoicesDefinition(
+        currentElementDefinition
+      );
+    }
 
-  if (isElementNodeChildOf(currentElementNode, previousElementNode)) {
     if (
-      previousElementNode.isArray &&
-      previousElementNode.type === "BackboneElement"
+      getKind(previousElementNodeDefinition.definition) === "choice" &&
+      previousElementNodeDefinition.childrenDefinitions.length > 0
     ) {
-      previousElementNode.backboneElementDefinitions.push(
-        currentElementDefinition
+      previousElementNodeDefinition.childrenDefinitions.forEach(
+        (prevChildDefinition) => {
+          prevChildDefinition.childrenDefinitions.push(currentDefinitionNode);
+          buildTreeDefinition(rest, rootNodeDefinition, currentDefinitionNode);
+        }
       );
-      buildTree(rest, rootNode, previousElementNode);
     } else {
-      previousElementNode.children.push(currentElementNode);
-      buildTree(rest, rootNode, currentElementNode);
+      previousElementNodeDefinition.childrenDefinitions.push(
+        currentDefinitionNode
+      );
+      buildTreeDefinition(rest, rootNodeDefinition, currentDefinitionNode);
     }
   } else {
-    const parent = getParent(previousElementNode, rootNode);
-    if (parent) buildTree(elementDefinitions, rootNode, parent);
+    const parent = getDefinitionNodeParent(
+      previousElementNodeDefinition,
+      rootNodeDefinition
+    );
+    if (parent)
+      buildTreeDefinition(elementDefinitions, rootNodeDefinition, parent);
```

Comment:
could you provide an example of ElementDefinition to illustrate each condition ? maybe not the whole EelementDefinition but at least the path or id

Reply:
it's not easy to understand what's going on without examples